### ISSUE TITLE
EmbedBuilder#setTitle does not allow a URL without a TLD (ex. localhost) if it has no file extension

### ISSUE THREAD

**Issue by queer** [DV8FromTheWorld/JDA#699]
[download]: https://bintray.com/dv8fromtheworld/maven/JDA/_latestVersion
[guild]: https://discord.gg/0hMr4ce0tIk3pSjp
[stack overflow]: https://stackoverflow.com/questions/tagged/java

## General Troubleshooting

Hey there! Before you report a bug or suggest a new feature,
  please make sure to follow these steps first!
  
- [x] I have checked for similar issues.
- [x] I have updated to the [latest JDA version][download].
- [x] I have checked the branches or the maintainers' PRs for upcoming features/bug fixes.

> This is not the place to learn Java. Please refer to [StackOverflow][stack overflow]
  for your general programming questions.

### Issue Type

- [x] Bug Report
- [ ] Feature Request

> For questions join the official JDA discord server: https://discord.gg/0hMr4ce0tIl3SLv5

### Description

Like title says. Attempting to build an embed with `EmbedBuilder` and setting the embed title URL to something like `http://localhost/whatever` fails. This is because the `URL_PATTERN` used expects some sort of TLD or file extension on the end.

https://github.com/DV8FromTheWorld/JDA/blob/3092875bd5fd1eec10f4973a34a13ab6b4170c6e/src/main/java/net/dv8tion/jda/core/EmbedBuilder.java#L43

Using a URL like this would be useful in the case of testing locally. I do know that you can use a hosts file entry like `127.0.0.1 localhost.localdomain` (and that's what I started doing), but it would be nice to not have to do this. 

Regex101 example showing this: https://regex101.com/r/A567ut/1/

**Comment by kantenkugel** [comment]
Not that i don't see your issue here, but i also don't see why localhost would make any sense here. Seeing as the only one that link would probably work on is you (the one sending the link)

**Comment by queer** [comment]
> Seeing as the only one that link would probably work on is you (the one sending the link)

Right, it's mainly just a convenience thing for doing local testing and not having to use some sort of alias like `localhost.localdomain`. 

> Additionally, im not even sure if discord would accept localhost urls (would be up to testing)

Discord does allow using `localhost.localdomain` as the URL at least when I tried it, so it seems logical that it'd also allow `localhost`.

**Comment by jagrosh** [comment]
I believe that this was tested when sending rich embeds became possible. However, I just tested this today and using a url such as https://localhost does indeed work now.
